src/test/java/teammates/test/cases/action/StudentCourseJoinAuthenticatedActionTest.java

```diff
@@ -26,14 +27,15 @@ protected String getActionUri() {
     }
 
     @BeforeClass
-    public void classSetup() {
+    public void classSetup() throws Exception {
         // extra test data used on top of typical data bundle
         removeAndRestoreDataBundle(dataBundle);
     }
 
     @Override
     @Test
     public void testExecuteAndPostProcess() throws Exception {
+        dataBundle = loadDataBundle("/StudentCourseJoinAuthenticatedTest.json");
```

Comment:
Load of dataBundle is required here as it is executed after access control and in accessControl we have changed dataBundle to typicaldataBundle